### PR TITLE
[Merged by Bors] - feat: Support podOverrides

### PR DESCRIPTION
**Pull request by razvan**
# Description

Blocked by: https://github.com/stackabletech/operator-rs/pull/616

Part of https://github.com/stackabletech/issues/issues/346

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
# Author
- [x] Changes are OpenShift compatible
- [x] CRD changes approved
- [x] Helm chart can be installed and deployed operator works
- [x] Integration tests passed (for non trivial changes)
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

**Comment by razvan** [comment]
```
--- FAIL: kuttl (1525.90s)
    --- FAIL: kuttl/harness (0.00s)
        --- PASS: kuttl/harness/spark-examples_openshift-true_spark-3.4.0-stackable0.0.0-dev (84.41s)
        --- PASS: kuttl/harness/spark-ny-public-s3_openshift-true_spark-3.4.0-stackable0.0.0-dev_s3-use-tls-true (184.67s)
        --- PASS: kuttl/harness/pyspark-ny-public-s3-image_openshift-true_spark-3.4.0-stackable0.0.0-dev_ny-tlc-report-0.1.0 (157.66s)
        --- PASS: kuttl/harness/spark-pi-private-s3_openshift-true_spark-3.4.0-stackable0.0.0-dev (115.93s)
        --- PASS: kuttl/harness/spark-pi-public-s3_openshift-true_spark-3.4.0-stackable0.0.0-dev (109.19s)
        --- PASS: kuttl/harness/smoke_openshift-true_spark-3.4.0-stackable0.0.0-dev_s3-use-tls-true (198.39s)
        --- FAIL: kuttl/harness/resources_openshift-true_spark-3.4.0-stackable0.0.0-dev (253.58s)
        --- PASS: kuttl/harness/logging_openshift-true_spark-3.4.0-stackable0.0.0-dev_ny-tlc-report-0.1.0 (433.86s)
        --- FAIL: kuttl/harness/pod_overrides_openshift-true_spark-3.4.0-stackable0.0.0-dev (999.59s)
        --- PASS: kuttl/harness/spark-history-server_openshift-true_spark-3.4.0-stackable0.0.0-dev_s3-use-tls-true (297.28s)
        --- PASS: kuttl/harness/pyspark-ny-public-s3_openshift-true_spark-3.4.0-stackable0.0.0-dev (163.10s)
FAIL
```

**Comment by razvan** [comment]
```
--- PASS: kuttl (1083.06s)
    --- PASS: kuttl/harness (0.00s)
        --- PASS: kuttl/harness/resources_openshift-true_spark-3.4.0-stackable0.0.0-dev (86.05s)
        --- PASS: kuttl/harness/smoke_openshift-true_spark-3.4.0-stackable0.0.0-dev_s3-use-tls-true (216.79s)
        --- PASS: kuttl/harness/pyspark-ny-public-s3_openshift-true_spark-3.4.0-stackable0.0.0-dev (167.66s)
        --- PASS: kuttl/harness/pod_overrides_openshift-true_spark-3.4.0-stackable0.0.0-dev (187.23s)
        --- PASS: kuttl/harness/spark-pi-private-s3_openshift-true_spark-3.4.0-stackable0.0.0-dev (117.71s)
        --- PASS: kuttl/harness/spark-history-server_openshift-true_spark-3.4.0-stackable0.0.0-dev_s3-use-tls-true (306.04s)
        --- PASS: kuttl/harness/spark-examples_openshift-true_spark-3.4.0-stackable0.0.0-dev (84.41s)
        --- PASS: kuttl/harness/spark-pi-public-s3_openshift-true_spark-3.4.0-stackable0.0.0-dev (109.89s)
        --- PASS: kuttl/harness/pyspark-ny-public-s3-image_openshift-true_spark-3.4.0-stackable0.0.0-dev_ny-tlc-report-0.1.0 (148.79s)
        --- PASS: kuttl/harness/spark-ny-public-s3_openshift-true_spark-3.4.0-stackable0.0.0-dev_s3-use-tls-true (184.84s)
        --- PASS: kuttl/harness/logging_openshift-true_spark-3.4.0-stackable0.0.0-dev_ny-tlc-report-0.1.0 (464.95s)
PASS
```

File: tests/templates/kuttl/resources/10-assert.yaml.j2 [hunk 1]
```diff
@@ -19,26 +19,20 @@ spec:
...
-    - name: vector
-{% endif %}
     - name: spark
       resources:
         limits:
           cpu: "1"
           memory: 1Gi
         requests:
-          cpu: 200m
+          cpu: "1"
```
Comment (sbernauer) [inline tests/templates/kuttl/resources/10-assert.yaml.j2:35]: Sorry for the dump question, but why did this change?

File: tests/templates/kuttl/resources/10-deploy-spark-app.yaml.j2 [hunk 2]
```diff
@@ -5,9 +5,6 @@ metadata:
   name: resources-crd
 spec:
   version: "1.0"
-{% if lookup('env', 'VECTOR_AGGREGATOR') %}
-  vectorAggregatorConfigMapName: vector-aggregator-discovery
-{% endif %}
```
Comment (sbernauer) [inline tests/templates/kuttl/resources/10-deploy-spark-app.yaml.j2:10]: Why did you remove logging to vector? (we try to log to log.t2.stackable.tech vector for all testcases to better debug tests)

**Review by sbernauer — CHANGES_REQUESTED** [review]

File: tests/templates/kuttl/resources/10-assert.yaml.j2 (hunk 1 above)
Comment (razvan) [reply tests/templates/kuttl/resources/10-assert.yaml.j2:35]: No, it was surprising to me too. It's because of [this](https://github.com/stackabletech/spark-k8s-operator/blob/ca32d38ebc6cac6852a5b1ac2289cd5c576fd8de/rust/crd/src/lib.rs#L757).

Those properties take precedence over the pod templates.

File: tests/templates/kuttl/resources/10-deploy-spark-app.yaml.j2 (hunk 2 above)
Comment (razvan) [reply tests/templates/kuttl/resources/10-deploy-spark-app.yaml.j2:10]: Ah, I haven't thought of that. I wanted to have cleaner tests.

File: tests/templates/kuttl/resources/10-assert.yaml.j2 (hunk 1 above)
Comment (sbernauer) [reply tests/templates/kuttl/resources/10-assert.yaml.j2:35]: Ahh forgot about that. Could we please add a comment like
`This resources are set from spark-submit according to the Spark config`

File: tests/templates/kuttl/resources/10-assert.yaml.j2 [hunk 3]
```diff
@@ -47,14 +41,11 @@ metadata:
...
-    - name: vector
-{% endif %}
     - name: spark
       resources:
         limits:
           cpu: "1"
           memory: 1Gi
         requests:
-          cpu: 250m
+          cpu: "1"
```
Comment (sbernauer) [inline tests/templates/kuttl/resources/10-assert.yaml.j2:50]: Could we please add a comment like
`This resources are set from spark driver according to the Spark config`

File: tests/templates/kuttl/resources/10-deploy-spark-app.yaml.j2 (hunk 2 above)
Comment (sbernauer) [reply tests/templates/kuttl/resources/10-deploy-spark-app.yaml.j2:10]: I would prefer to keep logging to e.g. collect the exceptions that occurred when the test failed

File: tests/templates/kuttl/resources/10-deploy-spark-app.yaml.j2 [hunk 4]
```diff
@@ -17,8 +17,6 @@ spec:
     spark.kubernetes.driver.pod.name: "resources-crd-driver"
     spark.kubernetes.executor.podNamePrefix: "resources-crd"
   job:
-    logging:
-      enableVectorAgent: {{ lookup('env', 'VECTOR_AGGREGATOR') | length > 0 }}
```
Comment (sbernauer) [inline tests/templates/kuttl/resources/10-deploy-spark-app.yaml.j2:21]: Is this a leftover?

Comment (razvan) [reply tests/templates/kuttl/resources/10-deploy-spark-app.yaml.j2]: yes. missed that.

File: tests/templates/kuttl/resources/10-assert.yaml.j2 [hunk 5]
```diff
@@ -11,6 +11,9 @@ metadata:
     app.kubernetes.io/managed-by: spark.stackable.tech_sparkapplication
 spec:
   containers:
+{% if lookup('env', 'VECTOR_AGGREGATOR') %}
+    - name: vector
+{% endif %}
```
Comment (sbernauer) [inline tests/templates/kuttl/resources/10-assert.yaml.j2:16]: You have changed to order of containers, I don't think this will work
This will however only be tested when the env `VECTOR_AGGREGATOR` is set, which happens on jenkins

File: tests/templates/kuttl/resources/12-assert.yaml.j2 [hunk 6]
```diff
@@ -11,6 +11,9 @@ metadata:
     app.kubernetes.io/managed-by: spark.stackable.tech_sparkapplication
 spec:
   containers:
+{% if lookup('env', 'VECTOR_AGGREGATOR') %}
+    - name: vector
+{% endif %}
```
Comment (sbernauer) [inline tests/templates/kuttl/resources/12-assert.yaml.j2:16]: Order change as well

**Review by sbernauer — CHANGES_REQUESTED** [review]

File: tests/templates/kuttl/resources/10-assert.yaml.j2 (hunk 5 above)
Comment (razvan) [reply tests/templates/kuttl/resources/10-assert.yaml.j2]: corrected

**Review by sbernauer — APPROVED** [review]
LGTM when custom test pass

**Comment by razvan** [comment]
https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/spark-k8s-operator-it-custom/90/
